src/ripple/app/main/GRPCServer.cpp

```diff
@@ -450,13 +450,16 @@ GRPCServer::run()
     }
 }
 
-GRPCServer::~GRPCServer()
+void
+GRPCServer::onStop()
 {
     if (running_)
     {
         impl_.shutdown();
         thread_.join();
```

Comment:
Once we're past the `join`, then we're no longer running.  We should note that in member variable `running_`.  So I suggest adding
```
running_ = false;
```
immediately below the call to `join`.